syncer/ddl.go

```diff
@@ -138,6 +154,24 @@ func (s *Syncer) splitAndFilterDDL(
 
 		// get real tableNames before apply block-allow list
 		if s.onlineDDL != nil {
+			if onlineDDLFinish && len(tables) > 1 {
+				if i == 0 && s.onlineDDL.TableType(tables[0].Name) == onlineddl.RealTable &&
+					s.onlineDDL.TableType(tables[1].Name) == onlineddl.TrashTable {
+					onlineDDLMatched = trashTable
+					// record trash table to give users information to check regex
+					tableRecords[trashTable] = tables[1].String()
+				}
+				if i == 1 && s.onlineDDL.TableType(tables[0].Name) == onlineddl.GhostTable &&
+					s.onlineDDL.TableType(tables[1].Name) == onlineddl.RealTable {
+					if onlineDDLMatched == allTable {
```

Comment:
why do we need to check `if onlineDDLMatched == allTable`?

seems if it matches `i == 1 && s.onlineDDL.TableType(tables[0].Name) == onlineddl.GhostTable && s.onlineDDL.TableType(tables[1].Name) == onlineddl.RealTable` it is enough to determine that the re of the onlineDDL is a successful match.

Reply:
Because `onlineDDLMatched` only has value if we only match `1` online ddl regex.
If we match both online ddl regex or none ddl regex, we both return `allTable` and a `nil` error.